### PR TITLE
Re-enable analyzers and fix stylecop issues

### PR DESCRIPTION
**Pull request by LucGenetier**
Re-enable .Net analyzers and StyleCop
Fix style code issues
Ensured working in debug & release modes 
- for VS 2019 and 2022
- for building & running tests in VS

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs:56]: >base [](http://example.com/codeflow?start=29&length=4)

nit - can we disable the rule and allow explicit 'base."?

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by MikeStall** [comment]
I approved - but the CI is still failing on these. This may be same the same issue that caused us to disable them with #862.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs:56]: I agree and will try fixing this

**Comment by LucGenetier** [comment]
The check build/pipeline will need to work before I merge.
Needs more investigations

---
In reply to: [1353909280](https://github.com/microsoft/Power-Fx/pull/913#issuecomment-1353909280) [](http://example.com/codeflow?ancestors=1353909280,1353909280)

File: src/libraries/Microsoft.PowerFx.Connectors/Microsoft.PowerFx.Connectors.csproj
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Microsoft.PowerFx.Connectors.csproj:36]: This change can break consumers.  When building Dataverse.EVal against this private change, I get this error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1605	Detected package downgrade: System.Collections.Immutable from 6.0.0 to 5.0.0. Reference the package directly from the project to select a different version. 
 Microsoft.PowerFx.Dataverse.Eval -> Microsoft.PowerFx.Core 0.2.3-preview.20221219-004 -> System.Collections.Immutable (>= 6.0.0) 
 Microsoft.PowerFx.Dataverse.Eval -> System.Collections.Immutable (>= 5.0.0)	Microsoft.PowerFx.Dataverse.Eval	D:\dev\temp\Power-Fx-Dataverse\src\PowerFx.Dataverse.Eval\Microsoft.PowerFx.Dataverse.Eval.csproj	1

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/Microsoft.PowerFx.Connectors.csproj:36]: Agreed. I've added a comment in this PR

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Microsoft.PowerFx.Core.csproj
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Microsoft.PowerFx.Core.csproj:29]: We need to get approval for this change from our partners. 
It breaks consumers.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs:753]: note that await/.Result can change behavior around exception handling. 
But probably fine in this case.

**Review by MikeStall — APPROVED** [review]

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**
